libraries/browser-tracker-core/src/helpers.ts

```diff
@@ -147,20 +146,14 @@ export function getReferrer(oldLocation?: string) {
   }
 
   try {
-    referrer = windowAlias.top.document.referrer;
-  } catch (e) {
-    if (windowAlias.parent) {
-      try {
-        referrer = windowAlias.parent.document.referrer;
-      } catch (e2) {
-        referrer = '';
-      }
+    if (windowAlias.top) {
+      return windowAlias.top.document.referrer;
+    } else if (windowAlias.parent) {
+      return windowAlias.parent.document.referrer;
     }
-  }
-  if (referrer === '') {
-    referrer = document.referrer;
-  }
-  return referrer;
+  } catch {}
+
+  return document.referrer;
```

Comment:
I remember rewriting this, it'd be nice to get your opinion that this is functionally the same.

Reply:
I think the only functional difference would be in case `windowAlias.top` was present but `windowAlias.top.document` was undefined – in that case the previous implementation would revert to the `windowAlias.parent` and the current would not. But that's a very theoretical situation, I couldn't find a case or reason why that would happen so I think it's good.